scanner/src/main/kotlin/scanners/scanoss/ScanOss.kt

```diff
@@ -37,14 +37,15 @@ import org.ossreviewtoolkit.scanner.PathScanner
 import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
 import org.ossreviewtoolkit.utils.core.createOrtTempDir
 
-class ScanOss(
+class ScanOss internal constructor(
     name: String,
     scannerConfig: ScannerConfiguration,
-    downloaderConfig: DownloaderConfiguration
+    downloaderConfig: DownloaderConfiguration,
+    private val config: ScanOssConfig
 ) : PathScanner(name, scannerConfig, downloaderConfig) {
     class Factory : AbstractScannerFactory<ScanOss>("SCANOSS") {
         override fun create(scannerConfig: ScannerConfiguration, downloaderConfig: DownloaderConfiguration) =
-            ScanOss(scannerName, scannerConfig, downloaderConfig)
+            ScanOss(scannerName, scannerConfig, downloaderConfig, ScanOssConfig.create(scannerConfig))
```

Comment:
IMO it's a bit odd to pass both `scannerConfig` and `ScanOssConfig.create(scannerConfig)` to `ScanOss` here, and I'd rather only pass `scannerConfig` and let `ScanOss` internally call `ScanOssConfig.create(scannerConfig)`.

Reply:
Fixed but please look at it again if it is to your taste because it has consequence on the property initialization.